vertx-pg-client/src/test/java/io/vertx/pgclient/PgConnectionTestBase.java

```diff
@@ -462,4 +463,22 @@ public void testTransactionAbort(TestContext ctx) {
       });
     }));
   }
+
+  @Test
+  public void testCloseConnectionFromDifferentContext(TestContext ctx) {
+    Async done = ctx.async(1);
+    connector.accept(ctx.asyncAssertSuccess(conn -> {
+      conn.query("SELECT 1").execute(ctx.asyncAssertSuccess(res -> {
+        ctx.assertEquals(1, res.size());
+        // schedule from another context
+        VertxInternal vertxInternal = (VertxInternal) vertx;
+        vertxInternal.createEventLoopContext().runOnContext(v -> {
+          conn.close(v2 -> {
+            done.complete();
+          });
+        });
+      }));
+    }));
+    done.await(2000);
```

Comment:
await is not needed, vertx-unit will wait for you

you need to use await when there is an action to take after

Reply:
ah I thought it was going to make the test fail-faster rather than being stuck if the callback is never called.